runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -943,16 +951,31 @@ J9::Z::TreeEvaluator::inlineStringCodingHasNegatives(TR::Node *node, TR::CodeGen
    generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_BRC, node, cFlowRegionEnd);
 
    generateS390LabelInstruction(cg, TR::InstOpCode::label, node, processOutOfRangeChar);
-   generateRIInstruction(cg, TR::InstOpCode::LGHI, node, returnReg, 1);
-
-   TR::RegisterDependencyConditions* dependencies = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(0, 7, cg);
+   if (isCountPositives)
+      {
+      generateRRRInstruction(cg, TR::InstOpCode::SRK, node, returnReg, lengthReg, numCharsLeftToProcess);
+      generateRIInstruction(cg, TR::InstOpCode::AHI, node, returnReg, -1);
+      generateVRScInstruction(cg, TR::InstOpCode::VLGV, node, outOfRangeCharIndexGR, outOfRangeCharIndex, generateS390MemoryReference(7, cg), 0);
+      generateRRInstruction(cg, TR::InstOpCode::AR, node, returnReg, outOfRangeCharIndexGR);
+      }
+   else
+      {
+      generateRIInstruction(cg, TR::InstOpCode::LGHI, node, returnReg, 1);
+      }
+   TR::RegisterDependencyConditions* dependencies = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(0, isCountPositives ? 10 : 9, cg);
+   if (isCountPositives)
+      {
+      dependencies->addPostConditionIfNotAlreadyInserted(outOfRangeCharIndexGR, TR::RealRegister::AssignAny);
+      }
+   dependencies->addPostConditionIfNotAlreadyInserted(lengthReg, TR::RealRegister::AssignAny);
```

Comment:
Change overall looks OK to me, but one thing I missed in previous PR, but may be something we can address here. 

Why did we use `addPostConditionIfNotAlreadyInserted` for all the registers. All new registers have been allocated in this function and we should know which one is added to dependency condition and which is not. Though this is functionally ok, it would check all the added registers in the dependency list everytime to see if it is added or not. Given that we know which registers are added, may be addPostCondition is sufficient.